docs: expand admin manual for the Activity app (#14306)
nextcloud/documentation: admin_manual/configuration_server/activity_configuration.rst

Activity app
============

> The Activity app tracks events across your Nextcloud instance and can
> notify users via the activity stream, email, and push notifications.

Reviewer: The opening is clearer than before, but *"tracks events across your Nextcloud instance"* still overgeneralizes slightly. The app displays supported activities rather than literally tracking every system event. To distinguish it from backend logging (like `admin_audit`) and emphasize its user-facing nature, I suggest softening the phrasing:

```suggestion
The Activity app records and summarizes user-visible events across your Nextcloud instance and can
notify users via the activity stream, email, and push notifications.
```

It is shipped and enabled by default.

.. figure:: ../images/activity-settings-admin.png

   *The Activity admin settings page showing default notification
   preferences for new accounts.*

Configuring your Nextcloud for the Activity app
------------------------------------------------

Enabling email notifications
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To send activity notification emails, a working
:doc:`email_configuration` is required.

> It is also recommended to configure the background job to ``Webcron``
> or ``Cron`` as described in :doc:`background_jobs_configuration`.
> The ``Ajax`` mode may delay or skip email delivery.

Reviewer: I happen to be working on an update to the Background Jobs / Cron chapter at the moment... I'd suggest this be tweaked to something like:

```suggestion
It is also recommended to configure the background job run mode to a system
run mode (``System Cron`` or ``systemd`` as described in :doc:`background_jobs_configuration`.
The ``Ajax`` and ``Webcron`` modes may delay or skip email delivery.
```

Maybe even make it more direct: *"For reliable delivery, configure the background job run mode to..."*

We basically have two categories of run modes:

- Web [AJAX, Webcron]
- System [Cron, Systemd]

The former category are the ones with the most severe limitations, particularly for sending out emails, dealing with expirations, etc.

Admin settings
^^^^^^^^^^^^^^

As an administrator, you can:

* Enable or disable notification emails globally using the
  **Enable notification emails** checkbox in **Settings** >
  **Administration** > **Activity**.
* Configure the default notification preferences for new accounts.
  Existing users keep their personal settings. The defaults apply
  to accounts created after the change.

Configuration reference
-----------------------

The following ``config.php`` options control Activity app behavior.

.. list-table::
   :header-rows: 1
   :widths: 30 15 55

   * - Option
     - Default
     - Description
   * - ``activity_expire_days``
     - ``365``
     - Number of days to retain activity records. A daily background

>        job deletes all activities older than this value. Set to ``0``
>        to disable expiration.

Reviewer: ```suggestion
       job deletes all activities older than this value. Minimum is ``1`` day.
```
There doesn't look to be a way to disable in this way. Code clamps at a minimum of `1`: https://github.com/nextcloud/activity/blob/be46767211b72dbca7a5da59c82dd97db3d4a82a/lib/Data.php#L428-L434

   * - ``activity_use_cached_mountpoints``
     - ``false``
     - When ``true``, activities in team folders and external storages

>        are generated for all users with access, not just the current

Reviewer: ```suggestion
       are generated for all users with access, not just the acting
```

For consistency with the troubleshooting section.

       user. See :ref:`label-activities-groupfolders` for details and
       caveats.
   * - ``activity_expire_exclude_users``
     - ``[]``
     - An array of user IDs whose activity records are never deleted
       by the expiration job. See :ref:`label-activities-exclude-users`.

.. _label-activities-groupfolders:

> Activities in team folders or external storages

Reviewer: ```suggestion
Activities in Team Folders or External Storages
```

-----------------------------------------------

By default, activities in team folders or external storages are only
generated for the current user. This is due to the underlying logic
of these storage backends. The config flag
``activity_use_cached_mountpoints`` makes activities work like in
normal shares when set to ``true``.

::

  'activity_use_cached_mountpoints' => true,

> .. warning::
> 
>    This config option comes with the following limitations:
> 
>    1. If "Advanced Permissions" (ACLs) are enabled in a team folder,
>       activities do not respect the permissions and all users see all
>       activities, even for files and directories they do not have
>       access to. **This potentially leaks sensitive information!** See
>       `this issue <https://github.com/nextcloud/groupfolders/issues/1057>`_
>       for more information.
>    2. Users that had access to a team folder, share or external
>       storage can see activities in their stream and emails that
>       happen after they are removed until they log in again.
>    3. Users that are newly added to a team folder, share or external
>       storage cannot see activities in their stream nor emails that
>       happen after they are added until they log in again.

Reviewer: The ACL information leak is materially more severe than the other two caveats, so it should stand on its own as a danger callout. The login-refresh behaviors are more like operational limitations, so they fit better as warning or note blocks.

I'd take advantage of this opportunity to split these up more clearly and use more appropriate callout types. Something like:

```suggestion
.. danger::

   If "Advanced Permissions" (ACLs) are enabled in a team folder,
   activities do not respect those permissions. As a result, users may
   see activity entries for files and directories they cannot access.
   **This can leak sensitive information.** See
   `this issue <https://github.com/nextcloud/groupfolders/issues/1057>`_
   for more information.

.. warning::

   Users who previously had access to a team folder, share, or external
   storage can continue to see new activity entries in their stream and
   emails until they log in again after access is removed.

.. note::

   Users who are newly added to a team folder, share, or external
   storage will not see new activity entries in their stream or emails
   until they log in again after access is granted.
```

.. _label-activities-exclude-users:

Excluding users from activity expiration
----------------------------------------

For certain users, it might make sense to never expire their activity
data, for example administrators. Set the config value
``activity_expire_exclude_users`` in your ``config.php``::

  'activity_expire_exclude_users' => [
    'admin',
    'group_admin',
    'second_admin'
  ]

For these users, their activity records will never be deleted from the
database.

> Better scheduling of activity emails
> -------------------------------------

Reviewer: ```suggestion
Better scheduling of activity emails
------------------------------------
```

2. Check that **Enable notification emails** is turned on in
   **Settings** > **Administration** > **Activity**.
3. Verify that the user has email notifications enabled in their
   personal notification settings (**Settings** > **Personal** >
   **Notifications**).
4. Ensure the background job is running. If you use ``Cron``, check
   that the system cron job is executing
   ``php -f /var/www/nextcloud/cron.php`` regularly.
5. Check the Nextcloud log (``data/nextcloud.log``) for mail-related
   errors.

Activities are missing for shared files or team folders
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

By default, activities in team folders and external storages are only
generated for the acting user. See
:ref:`label-activities-groupfolders` for how to enable activities for
all users with access.

Database growing large due to activity data
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The Activity app stores all events in the database. On instances with
many users or high file turnover, this can lead to significant
database growth. To manage this:

* Set ``activity_expire_days`` to a lower value (e.g. ``90`` or
  ``180``) to automatically clean up older records.

> * Monitor the ``oc_activity`` and ``oc_activity_mq`` tables directly

Reviewer: Maybe move the table bullet to the end to avoid inadvertently suggesting people mess with their database until exhausting other options. ;)

Nit: Not sure if ``activity_mq`` is important enough to mention (though it's also perfectly fine to do so admittedly). Isn't it more a transient data table?

  if you need to assess current size.
* Ensure the Nextcloud background job is running so that the daily

>   expiration job executes.

Reviewer: Maybe also mention the admin_audit app:

```suggestion
  expiration job executes.
  * If a non-user-facing audit trail is the primary goal (or auditing requires a different retention cycle), consider utilizing the ``admin_audit`` app.
```
